modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbStorage.java

```diff
@@ -121,30 +147,28 @@ public TxStateRocksDbStorage(
                 .putShort((short) partitionId)
                 .array();
 
-        byte[] indexAndTermBytes = readLastAppliedIndexAndTerm(readOptions);
-        lastAppliedIndex = indexAndTermBytes == null ? 0 : bytesToLong(indexAndTermBytes);
-        lastAppliedTerm = indexAndTermBytes == null ? 0 : bytesToLong(indexAndTermBytes, Long.BYTES);
-
-        persistedIndex = lastAppliedIndex;
+        initLastApplied();
     }
 
     @Override
     @Nullable
     public TxMeta get(UUID txId) {
         if (!busyLock.enterBusy()) {
-            throwStorageStoppedException();
+            throwExceptionIfStorageClosedOrRebalance();
```

Comment:
What's up with the rebalance in the exception if the storage is closed?

Reply:
I did not understand the question.
If we weren't able to take the "busy" lock, then it's either because the storage is closed or it started rebalancing.

Reply:
Why do we perform "busyLockSuccess" check if rebalance is covered by the busy lock itself?

Reply:
The "busy" lock only protects against the start of the rebalance, but not after.